src/components/ObsDetailsDefaultMode/SavedMatch/SavedMatchContainer.tsx

```diff
@@ -0,0 +1,40 @@
+import { useNavigation } from "@react-navigation/native";
+import { NativeStackNavigationProp } from "@react-navigation/native-stack";
+import SavedMatch from "components/ObsDetailsDefaultMode/SavedMatch/SavedMatch";
+import _ from "lodash";
+import React from "react";
+import type { RealmObservation } from "realmModels/types";
+
+type Props = {
+  observation: RealmObservation,
+}
+
+const SavedMatchContainer = ( {
+  observation
+}: Props ) => {
+  const navigation = useNavigation<NativeStackNavigationProp<Record<string, { id?: number }>>>( );
+
+  const navToTaxonDetails
+  = ( photo?: { id?: number; isRepresentativeButOtherTaxon?: boolean } ) => {
+    const navParams: {
+        id?: number;
+        firstPhotoID?: number;
+        representativePhoto?: unknown;
+    } = { id: observation.taxon?.id };
+    if ( !photo?.isRepresentativeButOtherTaxon ) {
```

Comment:
This entire if..else can be removed. The information about the representative_photo is only transiently available in the AI camera -> Match screen ( ->  Taxon Details) flow. It is not persisted to the realm DB and not available for such saved observations that we want to show the SavedMatch screen for. So, the only param we can send to TaxonDetails here is the taxon ID. And you can remove the photo argument.